### DISCUSSION TITLE
file upload

### DISCUSSION THREAD

**Discussion by NazarAli** [filamentphp/filament#8275] [Help] [answered]
in file upload how can upload multi files with store size and original file name. and when upload show the original file name

**Comment by leandrocfe** [comment, marked as answer]
```php
FileUpload::make('attachments')
    ->multiple()
    ->preserveFilenames()
```
[file-upload](https://filamentphp.com/docs/3.x/forms/fields/file-upload#uploading-multiple-files)

**Reply by NazarAli** [reply]
this just save original file name
i want save the size and extension. to give admin number of attachment and size of each type

**Reply by leandrocfe** [reply]
I think you can use [filament-spatie-media-library](https://filamentphp.com/plugins/filament-spatie-media-library#adding-custom-properties)
